actions/actions_whitebox_test.go

```diff
@@ -23,56 +22,60 @@ type ActionSuite struct {
 	gormtestsupport.DBTestSuite
 }
 
-func createWICopy(ID uuid.UUID, state string, boardcolumns []string) workitem.WorkItem {
+func createWICopy(wi workitem.WorkItem, state string, boardcolumns []interface{}) workitem.WorkItem {
 	var wiCopy workitem.WorkItem
-	wiCopy.ID = ID
-	fields := map[string]interface{}{
-		"system.state":        state,
-		"system.boardcolumns": boardcolumns,
+	wiCopy.ID = wi.ID
```

Comment:
Why don't you initialize the object right away?

```go
wiCopy := workitem.WorkItem {
  ID: wi.ID,
  SpaceID: wi.SpaceID,
  Type: wi.Type,
  Number: wi.Number,
  // ...
}
```

And then only deal with those values that cannot be set at once and require iteration.

Reply:
Also you might want to consider a deep copy library as described in the last section of [this article](https://flaviocopes.com/go-copying-structs/): https://github.com/jinzhu/copier. Then you can create a deep copy of the work item and only overwrite the sate and boardcolumns.

Reply:
This position is rather delicate. I had some issues there that were very fine granular. I changed it a bit, but left it mostly intact and very explicit.

Reply:
I understand that is delicate, as is the fact that copying should be as generic as possible and if we add something to the work item structure, we're likely to miss out on the copy function of yours.

Reply:
I don't think that makes sense at this point.